core/storage/blog_post/gae_models.py

```diff
@@ -0,0 +1,454 @@
+            'thumbnail_filename': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'published_on': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+        })
+
+    @classmethod
+    def create(cls, blog_post_id, author_id):
+        """Creates a new BlogPostSummaryModel entry.
+
+        Args:
+            blog_post_id: str. Blog Post ID of the newly-created blog post.
+            author_id: str. User ID of the author.
+
+        Returns:
+            BlogPostSummaryModel. The newly created BlogPostSummaryModel
+            instance.
+
+        Raises:
+            Exception. A blog post summary model with the given blog post ID
+                exists already.
+        """
+        if cls.get_by_id(blog_post_id):
+            raise Exception(
+                'Blog Post ID conflict on creating new blog post summary model.'
+                )
+
+        entity = cls(
+            id=blog_post_id,
+            author_id=author_id,
+            summary='',
+            title='',
+            published_on=None,
+            url_fragment='',
+            tags=[],
+            thumbnail_filename=''
+        )
+        entity.update_timestamps()
+        entity.put()
+
+        return entity
+
+    @classmethod
+    def get_blog_post_summary_models(cls, blog_post_ids):
+        """Returns a list of BlogPostSummaryModels for blog posts created by
+        the user.
+
+        Args:
+            blog_post_ids: list(str). Contains blog post ids for which the blog
+                post summary models are to be fetched.
+
+        Returns:
+            list(BlogPostSummaryModel). The list of BlogPostSummaryModels
+            fetched using blog_post_ids.
+        """
+        return cls.get_multi(blog_post_ids)
+
+
+class BlogPostRightsModel(base_models.BaseModel):
+    """Storage model for rights related to a blog post.
+
+    The id of each instance is the blog_post_id.
+    """
+
+    # The user_id of the blog editors of this blog post.
+    editor_ids = datastore_services.StringProperty(indexed=True, repeated=True)
+
+    # Whether this blog post is published or not.
+    # False if blog post is a draft, True if published.
+    blog_post_is_published = datastore_services.BooleanProperty(
+        indexed=True, required=True, default=False)
+
+    @staticmethod
+    def get_deletion_policy():
+        """Model contains data to pseudonymize or delete corresponding
+        to a user: editor_ids field.
+        """
+        return base_models.DELETION_POLICY.LOCALLY_PSEUDONYMIZE
+
+    @classmethod
+    def has_reference_to_user_id(cls, user_id):
+        """Check whether BlogPostRightsModel references user.
+
+        Args:
+            user_id: str. The ID of the user whose data should be checked.
+
+        Returns:
+            bool. Whether any models refer to the given user ID.
+        """
+        return cls.query(
+            # This checks if any of the editor IDs is equal to the user_id.
```

Comment:
This comment seems a bit redundant to me.

Reply:
Brian asked to add the comment as it is not that clear. 
https://github.com/oppia/oppia/pull/12933#discussion_r641015242

Reply:
Please add a bit more context:
```suggestion
            # NOTE: Even though `editor_ids` is repeated, we can compare it to a single value and it will return models where any of the editor IDs are equal to user_id.
```

Reply:
Added.